### PR TITLE
Rename internal `Iterator::Slice` type to not conflict with `::Slice`

### PR DESCRIPTION
**Pull request by Blacksmoke16**
Ref: https://forum.crystal-lang.org/t/compilation-error-iterator-bug/5276

### CODE REVIEW

File: spec/std/iterator_spec.cr [hunk 1]
```diff
@@ -527,6 +541,10 @@ describe Iterator do
       iter.next.should eq([7, 8])
       iter.next.should be_a(Iterator::Stop)
     end
+
+    it "doesnt conflict with `::Slice` type" do
+      assert_iterates_iterator [1, 2, 3], MockIterator.new.each
+    end
```
Comment (Sija) [inline spec/std/iterator_spec.cr:547]: Does it makes sense, actually?

Comment (Blacksmoke16) [reply spec/std/iterator_spec.cr:547]: Probably a bit overkill yea, but doesn't really hurt anything either. Maybe it'll prevent a regression in crystal 3.0 😅

**Comment by straight-shoota** [comment]
I think we should do the same for all iterators defined in `Iterator`. Other iterators defined outside of `Iterator` have an iterator "Iterator" suffix which should go a great way to make random name collisions unlikely.

**Review by straight-shoota — APPROVED** [review]

**Comment by straight-shoota** [comment]
Failing specs need attention

**Comment by Blacksmoke16** [comment]
Ogh well that's lovely. There is `Iterator::Step` and `Steppable::StepIterator` of which `Steppable` includes `Iterator` so now we have two `StepIterators`...

**Comment by Blacksmoke16** [comment]
Okay so apparently `Step::StepIterator` is a public type so can't really rename that one. Imma just give the other internal one a diff name and call it a day.

**Review by straight-shoota — APPROVED** [review]

**Pull request merged**